### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by Yazino12**
### In this pull:

**Set up associations between models.**

- [Create model classes for all entities](https://github.com/Yazino12/blog-app/pull/2/commits/fd6fd549dd376cbeb780beeb7359d1e592b9df1a)

- [Set up associations between models](https://github.com/Yazino12/blog-app/pull/2/commits/9aca5d84a9805a78aa5d7dc7d80ed8124344c225)

- [Write all record creation to seeds.rb](https://github.com/Yazino12/blog-app/pull/2/commits/11cc4cea804bab91e8ca89a4fba523b94bb8f92d)

- [Create custom methods for the models](https://github.com/Yazino12/blog-app/pull/2/commits/13aaabb27f5ef58310c1d49af71b8cee42d2d289)

- [Run Rubocop and resolve errors](https://github.com/Yazino12/blog-app/pull/2/commits/a5e6b53c4293caa1062ee313dca96ffd741929b5)

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comment_counter
+    post.increment!(:comment_counter)
+  end
```
Comment (zainsadaqat) [inline app/models/comment.rb]: #### Optional
*you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -* `after_save` to call the `#update_comment_counter` method every time a new post has been saved to database, then make the `update_comment_counter` method private

```ruby

  after_save :update_comment_counter

  private

  def update_comment_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_like_counter
+    post.increment!(:like_counter)
+  end
```
Comment (zainsadaqat) [inline app/models/like.rb:11]: #### Optional 
*you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -* `after_save` to call the `#update_like_counter` method every time a new post has been saved to database, then make the `update_like_counter` method private

```ruby

  after_save :update_like_counter

  private

  def update_like_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_post_counter
+    author.increment!(:post_counter)
+  end
```
Comment (zainsadaqat) [inline app/models/post.rb]: #### Optional

*you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -* `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private

```ruby

  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @Yazino12  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 3 above)
Comment (Yazino12) [reply app/models/post.rb]: @zainsadaqat, Thanks for taking the time and for the extra suggestions 🙏🏾

**Pull request merged**